src/annotator.cpp

```diff
@@ -812,30 +827,30 @@ bool Annotator::assignAllIds(ModelPtr &model)
         issue->mPimpl->setReferenceRule(Issue::ReferenceRule::ANNOTATOR_NULL_MODEL);
         return false;
     }
+    auto curModel = this->model();
```

Comment:
Documentation says that this should overwrite any existing model in this Annotator. I think we should stick with the documented approach and not cache the current model.

Reply:
[d32b338](https://github.com/cellml/libcellml/pull/1151/commits/d32b3387c5ceba5aacc8a923de17f3db67f6062a)